src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -106,22 +111,40 @@ public Query rewrite(IndexSearcher indexSearcher) throws IOException {
             return new MatchNoDocsQuery("empty HybridQuery");
         }
 
-        boolean actuallyRewritten = false;
-        List<Query> rewrittenSubQueries = new ArrayList<>();
+        HybridQueryRewriteCollectorManager manager = new HybridQueryRewriteCollectorManager(indexSearcher);
+        List<Callable<Void>> queryRewriteTasks = new ArrayList<>();
+        List<HybridQueryExecutorCollector<IndexSearcher, Map.Entry<Query, Boolean>>> collectors = new ArrayList<>();
```

Comment:
lets make all the variable final here. We should not allow anyone to change references of these variables.

Reply:
Ack